cfme/tests/services/test_orchestration_template.py

```diff
@@ -48,82 +49,137 @@
-        'Error during Orchestration Template creation: new template content cannot be empty'
-    template_type = provisioning['stack_provisioning']['template_type']
-    temp_type = provisioning['stack_provisioning']['template_type_dd']
+        "Error during Orchestration Template creation: new template content cannot be empty"
     collection = appliance.collections.orchestration_templates
     with error.expected(flash_msg):
-        collection.create(template_type=template_type, temp_type=temp_type,
+        collection.create(template_group=templates.get(template_type)[1],
+                          template_type=templates.get(template_type)[0],
                           template_name=fauxfactory.gen_alphanumeric(),
-                          description="my template", content='')
-
-
-def test_tag_orchestration_template(appliance, provisioning, tag, create_template):
-    template_type = provisioning['stack_provisioning']['template_type']
-    temp_type = provisioning['stack_provisioning']['template_type_dd']
+                          description="my template",
+                          content='')
+
+
+def test_tag_orchestration_template(tag, created_template):
+    """Tests template tagging. Verifies that tag was added, confirms in template details,
+    removes tag
+    """
+    navigate_to(created_template, 'Details')
+    created_template.add_tag(tag=tag)
+    assert ((tag.display_name, tag.category.display_name) in
+            [(tags.display_name, tags.category.display_name) for tags in
+             created_template.get_tags()])
+    created_template.remove_tag(tag=tag)
+
+
+@pytest.mark.parametrize("action", ["copy", "create"], ids=["copy", "create"])
+def test_duplicated_content_error_validation(appliance, created_template, template_type,
+                                             action):
+    """Tests that we are not allowed to have duplicated content in different templates"""
     collection = appliance.collections.orchestration_templates
-    template = collection.create(template_type=template_type, temp_type=temp_type,
-                                 template_name=fauxfactory.gen_alphanumeric(),
-                                 description="my template", content=create_template)
-    navigate_to(template, 'Details')
-    template.add_tag(tag=tag)
-    template.remove_tag(tag=tag)
-    template.delete()
+    if action == "copy":
+        copy_name = created_template.template_name + "_copied"
+        flash_msg = ("Unable to create a new template copy \"{}\": old and new template content "
+                     "have to differ.".format(copy_name))
+        with error.expected(flash_msg):
+            created_template.copy_template(copy_name, created_template.content)
+    elif action == "create":
+        flash_msg = "Error during 'Orchestration Template creation': " \
+                    "Validation failed: Md5 of content already exists (content must be unique)"
+        with error.expected(flash_msg):
```

Comment:
**optional** please use `pytest.raises(..., match=regex_for_mesage)`

Reply:
@ronny that's a nice idea. But i think I'll add this to `cfme.utils.error` to avoid direct pytest calls in tests. right?

Reply:
@lkhomenk i'm going to remove `utils.error` in #6994